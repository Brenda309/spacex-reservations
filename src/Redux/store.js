import { configureStore } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import reducerMissions from './mission';

const rootReducer = reducerMissions;

const store = configureStore({
  reducer: {
    missions: rootReducer,
  },
  applyMiddleware: [thunk],
});

export default store;
