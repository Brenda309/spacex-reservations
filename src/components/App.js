/* eslint-disable import/no-named-as-default */
import React from 'react';
import { Routes, Route } from 'react-router-dom';
import DragonPage from './Dragon/DragonPage';
import MissionPage from './Mission/MissionPage';
import RocketsPage from './Rockets/RocketsPage';
import ProfilePage from './ProfilePage';
import './App.css';

function App() {
  return (
    <>
      <Routes>
        <Route
          exact
          path="/"
          element={(<ProfilePage />)}
        />

        <Route
          path="/dragons"
          element={<DragonPage />}
        />

        <Route
          path="/missions"
          element={<MissionPage />}
        />

        <Route
          path="/rockets"
          element={<RocketsPage />}
        />
      </Routes>
    </>
  );
}

export default App;
