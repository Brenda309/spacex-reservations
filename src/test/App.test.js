import renderer from 'react-test-renderer';
import {
  render, screen, waitFor,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/configState';
import App from '../components/App';

describe('Body of the whole Rocket Page', () => {
  it('Check Falcon 1 in the DOM', async () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </MemoryRouter>,
    );
    await waitFor(() => {
      const waitForDOM = screen.findByText(
        /The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth./,
      );
      expect(waitForDOM).not.toBeNull();
    });
  });

  it('Check Falcon 9 in the DOM', async () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </MemoryRouter>,
    );
    await waitFor(() => {
      const waitForDOM = screen.findByText(
        /Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit./,
      );
      expect(waitForDOM).not.toBeNull();
    });
  });

  it('Check Falcon Heavy in the DOM', async () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </MemoryRouter>,
    );
    await waitFor(() => {
      const waitForDOM = screen.findByText(
        /With the ability to lift into orbit over 54 metric tons (119,000 lb)--a mass equivalent to a 737 jetliner loaded with passengers, crew, luggage and fuel--Falcon Heavy can lift more than twice the payload of the next closest operational vehicle, the Delta IV Heavy, at one-third the cost./,
      );
      expect(waitForDOM).not.toBeNull();
    });
  });

  it('Check Starship in the DOM', async () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </MemoryRouter>,
    );
    await waitFor(() => {
      const waitForDOM = screen.findByText(
        /Starship and Super Heavy Rocket represent a fully reusable transportation system designed to service all Earth orbit needs as well as the Moon and Mars. This two-stage vehicle — composed of the Super Heavy rocket (booster) and Starship (ship) — will eventually replace Falcon 9, Falcon Heavy and Dragon./,
      );
      expect(waitForDOM).not.toBeNull();
    });
  });

  it('It matches snapshot', () => {
    const Tree = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </MemoryRouter>,
    ).toJSON();
    expect(Tree).toMatchSnapshot();
  });
});
