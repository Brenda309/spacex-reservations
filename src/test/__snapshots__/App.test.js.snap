// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body of the whole Rocket Page It matches snapshot 1`] = `
<div>
  <header
    className="container"
  >
    <div
      className="logo"
    >
      <img
        alt="logo"
        src="planet.png"
      />
      <h1>
        Space Travelers' hub
      </h1>
    </div>
    <nav>
      <ul
        className="list"
      >
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
            style={
              Object {
                "fontWeight": "bold",
                "textDecoration": "underline",
              }
            }
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            className=""
            href="/missions"
            onClick={[Function]}
            style={
              Object {
                "fontWeight": "300",
                "textDecoration": "none",
              }
            }
          >
            Missions
          </a>
        </li>
        <li
          className="dragons"
        >
          <a
            className=""
            href="/dragons"
            onClick={[Function]}
            style={
              Object {
                "fontWeight": "300",
                "textDecoration": "none",
              }
            }
          >
            Dragons
          </a>
        </li>
        <li>
          <a
            className=""
            href="/profile"
            onClick={[Function]}
            style={
              Object {
                "fontWeight": "300",
                "textDecoration": "none",
              }
            }
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <div>
    <header
      className="container"
    >
      <div
        className="logo"
      >
        <img
          alt="logo"
          src="planet.png"
        />
        <h1>
          Space Travelers' hub
        </h1>
      </div>
      <nav>
        <ul
          className="list"
        >
          <li>
            <a
              aria-current="page"
              className="active"
              href="/"
              onClick={[Function]}
              style={
                Object {
                  "fontWeight": "bold",
                  "textDecoration": "underline",
                }
              }
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              className=""
              href="/missions"
              onClick={[Function]}
              style={
                Object {
                  "fontWeight": "300",
                  "textDecoration": "none",
                }
              }
            >
              Missions
            </a>
          </li>
          <li
            className="dragons"
          >
            <a
              className=""
              href="/dragons"
              onClick={[Function]}
              style={
                Object {
                  "fontWeight": "300",
                  "textDecoration": "none",
                }
              }
            >
              Dragons
            </a>
          </li>
          <li>
            <a
              className=""
              href="/profile"
              onClick={[Function]}
              style={
                Object {
                  "fontWeight": "300",
                  "textDecoration": "none",
                }
              }
            >
              My Profile
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <div
      className="profile-sect"
    >
      <h1
        className="header"
      >
        My Missions
      </h1>
      <ul
        className="missionsBooked"
      >
        <li
          className="hide"
          style={
            Object {
              "border": "none",
            }
          }
        >
          reserved missions currently(0)
        </li>
      </ul>
    </div>
  </div>
</div>
`;
